assets/js/SignGroups.tsx

```diff
@@ -1,7 +1,361 @@
 import * as React from 'react';
+import ReactDatePicker from 'react-datepicker';
+import { stationConfig, arincToRealtimeId } from './mbta';
+import VirtualSign from './VirtualSign';
+import CustomTextInput from './CustomTextInput';
+import AlertPicker from './AlertPicker';
+import {
+  RouteAlerts,
+  StationConfig,
+  Zone,
+  RouteSignGroups,
+  SignGroup,
+} from './types';
+import { defaultZoneLabel } from './helpers';
 
-function SignGroups(): JSX.Element | null {
-  return <div>Sign Groups Here</div>;
+interface ZoneSelectorProps {
+  config: StationConfig;
+  zone: Zone;
+  line: string;
+  selectedSigns: { [sign: string]: boolean };
+  onSignChange: (signId: string, selected: boolean) => void;
+  readOnly: boolean;
+  kind: 'default' | 'named';
+}
+
+function ZoneSelector({
```

Comment:
Can we keep things to one component per file? This is a very long file as it is, and it's hard to follow what might be relevant for each component.

Reply:
Hmm, I'm ambivalent. My mental model is that this file is a module, and react components are just functions in a module. And so these non-exported ones are the equivalent of private functions, with all that entails regarding testability, freedom to change, etc. I think of `ZoneSelector` the same as if there was a little private helper function in here, like `shouldShowZone` or something. And I think of `SignGroups` as the single, shared component that the app uses.

I recognize that the file is pretty long, though. I feel like the react way of including markup sort of makes that unavoidable.

So what's the general principle we should have here? One react component per file? It doesn't matter to me that much, since I mostly just F12 to jump component to component. My only concern is that moving these private components to their own files means I have to export them, and then we'd test them, and then even though they're conceptually private we now have an implementation locked down. But since react components should _in principle_ be used anywhere, maybe that's okay. As part of this PR, for example, I did have to extract out two components that were previously private into their own public modules.

Reply:
Hm, you're right that strict "one component per file" doesn't leave a lot of room for privacy of implementation details in the React world. I guess it's a balance we have to strike, like not having an excessive depth/complexity of private functions in an Elixir module. I think this is probably fine, my eyes just started to glaze over the first time I tried to read it, but it's more difficult to scan code on GitHub than in an editor.